rfq/negotiator.go

```diff
@@ -625,6 +625,115 @@ func (n *Negotiator) HandleIncomingBuyAccept(msg rfqmsg.BuyAccept,
 	}()
 }
 
+// HandleIncomingSellAccept handles an incoming sell accept message. This method
+// is called when a peer accepts a quote request from this node. The method
+// checks the price and expiry time of the quote accept message. Once validation
+// is complete, the finalise callback function is called.
+func (n *Negotiator) HandleIncomingSellAccept(msg rfqmsg.SellAccept,
+	finalise func(rfqmsg.SellAccept, fn.Option[InvalidQuoteRespEvent])) {
+
+	// Ensure that the quote expiry time is within acceptable bounds.
+	//
+	// TODO(ffranr): Sanity check the quote expiry timestamp given
+	//  the expiry timestamp provided by the price oracle.
+	if !expiryWithinBounds(msg.Expiry, minRateTickExpiryLifetime) {
+		// The expiry time is not within the acceptable bounds.
+		log.Debugf("Sell accept quote expiry time is not within "+
+			"acceptable bounds (expiry=%d)", msg.Expiry)
+
+		// Construct an invalid quote response event so that we can
+		// inform the peer that the quote response has not validated
+		// successfully.
+		invalidQuoteRespEvent := NewInvalidQuoteRespEvent(
+			&msg, InvalidExpiryQuoteRespStatus,
+		)
+		finalise(
+			msg, fn.Some[InvalidQuoteRespEvent](
+				*invalidQuoteRespEvent,
+			),
+		)
+
+		return
+	}
+
+	if n.cfg.SkipAcceptQuotePriceCheck {
```

Comment:
Since we allow the price oracle to be `nil` now, we'll run into a panic if the user hasn't set the flag but also hasn't configured the oracle (they just ran stuff with the minimum set they could get things to start with, which is what we need to expect).

In my tests I added the following:
```
	// Reject the quote request if a price oracle is unavailable.
	if n.cfg.PriceOracle == nil {
		invalidQuoteRespEvent := NewInvalidQuoteRespEvent(
			&msg, PriceOracleQueryErrQuoteRespStatus,
		)
		finalise(msg, fn.Some[InvalidQuoteRespEvent](
			*invalidQuoteRespEvent,
		))
		return
	}
```

Which works quite nicely. Though it makes me wish the invalid quote also had an error string that we could give a bit more context or an error message to the user with.

Reply:
Ah, the same on the `HandleIncomingSellAccept` side, I guess.

Reply:
> Though it makes me wish the invalid quote also had an error string that we could give a bit more context or an error message to the user with.